dbms/src/main/java/org/polypheny/db/view/MaterializedManagerImpl.java

```diff
@@ -0,0 +1,441 @@
+                    if ( numberUpdated == (materializedCriteria.getInterval() - 1) ) {
+                        isUpdatingMaterialized = true;
+                        prepareToUpdate( view.id );
+                        updateMaterializedUpdate( view.id, 0 );
+                        isUpdatingMaterialized = false;
+                    } else {
+                        updateMaterializedUpdate( view.id, numberUpdated + 1 );
+                    }
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Register the freshnessLoop as BackgroundTask to update interval Materialized Views after a given time
+     */
+    private void registerFreshnessLoop() {
+        BackgroundTaskManager.INSTANCE.registerTask(
+                MaterializedManagerImpl.this::updatingIntervalMaterialized,
+                "Update Materialized View with freshness type interval if it is time.",
+                TaskPriority.HIGH,
+                (TaskSchedulingType) RuntimeConfig.FRESHNESSLOOP.getEnum() );
+    }
+
+
+    /**
+     * Update Materialized Views with freshness type interval if it is time to update them
+     */
+    private void updatingIntervalMaterialized() {
+        Map<Long, MaterializedCriteria> materializedViewInfo;
+        materializedViewInfo = ImmutableMap.copyOf( updateMaterializedViewInfo() );
+        materializedViewInfo.forEach( ( k, v ) -> {
+            if ( v.getCriteriaType() == CriteriaType.INTERVAL ) {
+                if ( v.getLastUpdate().getTime() + v.getTimeInMillis() < System.currentTimeMillis() ) {
+                    if ( !isDroppingMaterialized && !isCreatingMaterialized && !isUpdatingMaterialized ) {
+                        prepareToUpdate( k );
+                        updateMaterializedTime( k );
+                    }
+                }
+            }
+        } );
+    }
+
+
+    /**
+     * starts transition and acquires a global schema lock before the materialized view is updated
+     *
+     * @param materializedId of materialized view, which is updated
+     */
+    public void prepareToUpdate( Long materializedId ) {
+        Catalog catalog = Catalog.getInstance();
+        CatalogTable catalogTable = catalog.getTable( materializedId );
+
+        try {
+            Transaction transaction = getTransactionManager().startTransaction( catalogTable.ownerName, catalog.getDatabase( catalogTable.databaseId ).name, false, "Materialized View" );
+
+            if ( !(LockManager.INSTANCE.hasLock( (TransactionImpl) transaction, LockManager.GLOBAL_LOCK )) ) {
+                try {
+                    // Get a exclusive global schema lock
+                    LockManager.INSTANCE.lock( LockManager.GLOBAL_LOCK, (TransactionImpl) transaction, LockMode.EXCLUSIVE );
+                } catch ( DeadlockException e ) {
+                    throw new RuntimeException( "DeadLock while locking for materialized view update", e );
+                }
```

Comment:
Do we really need a global lock?

Reply:
Now using same logic as in AbstractQueryProcessor.